engine-src/Game/LambdaHack/Client/UI/HandleHumanGlobalM.hs

```diff
@@ -977,43 +940,40 @@ alterTile ts dir = do
 -- the action. Consequently, even if all embedded items are recharching,
 -- the time will be wasted and the server will describe the failure in detail.
 alterTileAtPos :: MonadClientUI m
-               => [TriggerTile] -> Point -> Text
+               => Maybe Point
                -> m (FailOrCmd RequestTimed)
-alterTileAtPos ts tpos pText = do
-  cops@COps{cotile, coTileSpeedup} <- getsState scops
-  leader <- getLeaderUI
-  b <- getsState $ getActorBody leader
-  actorSk <- leaderSkillsClientUI
-  lvl <- getLevel $ blid b
-  embeds <- getsState $ getEmbedBag (blid b) tpos
-  let alterSkill = Ability.getSk Ability.SkAlter actorSk
-      t = lvl `at` tpos
-      alterMinSkill = Tile.alterMinSkill coTileSpeedup t
-      hasFeat TriggerTile{ttfeature} = Tile.hasFeature cotile ttfeature t
-  case filter hasFeat ts of
-    [] | not $ null ts -> failWith $ guessAlter cops ts t
-    _ | not (Tile.isModifiable coTileSpeedup t)
-        && EM.null embeds -> failSer AlterNothing
-    _ | chessDist tpos (bpos b) > 1 -> failSer AlterDistant
-    _ | alterSkill <= 1 -> failSer AlterUnskilled
-    _ | not (Tile.isSuspect coTileSpeedup t)
-        && alterSkill < alterMinSkill -> failSer AlterUnwalked
-    trs ->
-      if EM.notMember tpos $ lfloor lvl then
-        if not (occupiedBigLvl tpos lvl)
-           && not (occupiedProjLvl tpos lvl) then do
-          let v = case trs of
-                [] -> "alter"
-                tr : _ -> ttverb tr
-          verAlters <- verifyAlters (blid b) tpos
-          case verAlters of
-            Right () -> do
-              let msg = makeSentence ["you", v, MU.Text pText]
-              msgAdd MsgDone msg
-              return $ Right $ ReqAlter tpos
-            Left err -> return $ Left err
-        else failSer AlterBlockActor
-      else failSer AlterBlockItem
+alterTileAtPos mp = case mp of
+  Nothing -> failWith "never mind"
+  Just tpos -> do
+    COps{coTileSpeedup} <- getsState scops
+    leader <- getLeaderUI
+    b <- getsState $ getActorBody leader
+    actorSk <- leaderSkillsClientUI
+    lvl <- getLevel $ blid b
+    embeds <- getsState $ getEmbedBag (blid b) tpos
+    let alterSkill = Ability.getSk Ability.SkAlter actorSk
+        t = lvl `at` tpos
+        alterMinSkill = Tile.alterMinSkill coTileSpeedup t
+    if | not (Tile.isModifiable coTileSpeedup t) && EM.null embeds
+        -> failSer AlterNothing
+       | chessDist tpos (bpos b) > 1
+        -> failSer AlterDistant
+       | alterSkill <= 1
+        -> failSer AlterUnskilled
+       | not (Tile.isSuspect coTileSpeedup t) && alterSkill < alterMinSkill
+        -> failSer AlterUnwalked
+       | otherwise  
+        -> if EM.notMember tpos $ lfloor lvl then
```

Comment:
Now, after your simplifications, you can fold these ifs into the big if above (negating the conditions).